File: PrintAsciiPyramid/pyramid.go
package main
//import "os"
func main(){
//	print(pyramid(os.Args[1]))
	println(pyramid(`//\\/\\/\\\\`))
}

func pyramid (a string) string {
	o,b,e,w := a,2,len(a)-2, " "
	for b<=e {
		a := []byte(a)[b:e]
		for i,_ := range a { 
			if a[i]=='/' {
				a[i]='\\'
			} else {
				a[i]='/'
			} 
		}
		o = w + "/" + string(a) + "\\\n" + o
		w+=" "
		b++
		e--
	}
	return o
}

/* // Old Version
package main

func main() {

	const arg = `//\\/\\/\\\\`
	const width = len(arg)
	const height = width/2
	
	var a [height][width]byte
	// fmt.Println(len(arg))
	
	for y,row := range a {
		for x,_ := range row {
			a[y][x] = ' '
		}
	}
	// for _,a := range a { fmt.Println(string(a[:])) }
	
	for x,_ := range a[height-1] {
		a[height-1][x] = arg[x]
	}

	b := 1
	e := width-2


// tests
//	a[0][b]=1
//	a[0][e]=2
//	a[height-1][width-1]=99
//	fmt.Println(a)
	
	for y:=height-1; y>=1; y-- {
		a[y-1][b] = '/'
		for x:=b+1; x<=e-1; x++ {
			if a[y][x] == '/' { 
				a[y-1][x] = '\\'
			}
			if a[y][x] == '\\' { 
				a[y-1][x] = '/'
			}		
		}
		a[y-1][e] = '\\'
		b++
		e--
	}
	
	for y,row := range a {
		for x,_ := range row {
			print(string(a[y][x]))
		}
		println()
	}
	
}

*/

File: PrintAsciiPyramid/pyramid.c
#include <stdio.h>
#include <string.h>
int asciiPyramid(char* a);

int main(){

	int i;
	const int N = 4;

	char* tests[N] = {
		"//\\\\/\\\\/\\\\\\\\",
		"/\\\\\\\\\\\\",
		"//\\\\\\\\///\\\\/\\\\",
		"/////////////\\\\",
	};

	for(i=0;i<=N-1;i++){
		asciiPyramid(tests[i]);
	}
}
int asciiPyramid(char* a){

	// char a[] = "//\\\\/\\\\/\\\\\\\\";
	
	int width = strlen(a);
	// make width even
	if (width%2==1) width--;
	int height = width/2;
	char table[height][width];

	int x,y;
	int beg = 1;
	int end = width-2;
	// beginning b and end e of line


	// initialize to ' '
	for(y=0; y<=height-1; y++)
		for(x=0; x<=width-1; x++)
			table[y][x] = ' ';
			
	// set last row to argument
	for(x=0; x<=width-1; x++){
		table[height-1][x] = a[x];
	}

	// make pyramid
	for(y=height-1; y>=1; y--){
		table[y-1][beg] = '/';
		for (x=beg+1; x<=end-1; x++) {
			if (table[y][x] == '/') { 
				table[y-1][x] = '\\';
			}
			if (table[y][x] == '\\') { 
				table[y-1][x] = '/';
			}		
		}
		table[y-1][end] = '\\';
		beg++;
		end--;
	}

	// print
	for(y=0; y<=height-1; y++){
		for(x=0; x<=width-1; x++){
			printf("%c", table[y][x]);
		}
		printf("\n");
	}
	
	return 0;
}